modules/test/playwright/tests/object-web/objectFolder.spec.ts

```diff
@@ -12,43 +12,123 @@ import {getRandomInt} from '../../utils/getRandomInt';
+		apiHelpers,
+		modelBuilderPage,
+	}) => {
+		const objectFolders: ObjectFolder[] = await Promise.all(
+			Array.apply(null, Array(5)).map(async () => {
+				return await apiHelpers.objectAdmin.postRandomObjectFolder();
+			})
+		);
 
-	const objectFolderExternalReferenceCode = 'objectFolder' + getRandomInt();
+		await modelBuilderPage.goto({objectFolderName: 'Default'});
 
-	const objectFolder = await viewObjectDefinitionsPage.createObjectFolder(
-		objectFolderExternalReferenceCode
-	);
+		for (const objectFolder of objectFolders) {
+			await expect(modelBuilderPage.otherObjectFolders).toBeVisible();
 
-	await expect(
-		viewObjectDefinitionsPage.page
-			.locator('li')
-			.filter({hasText: objectFolderExternalReferenceCode})
-	).toBeVisible();
+			const otherObjectFolderLocator =
+				modelBuilderPage.otherObjectFolderLocator({
+					objectFolderLabel: objectFolder.label['en_US'],
+				});
 
-	// Clean up
+			await otherObjectFolderLocator.hover();
 
-	await apiHelpers.objectAdmin.deleteObjectFolder(objectFolder.id);
+			await otherObjectFolderLocator
+				.getByRole('button', {name: 'Go to Folder'})
+				.click();
+
+			await expect(otherObjectFolderLocator).toBeHidden();
+
+			await expect(
+				modelBuilderPage.objectFolderLabelHeaderLocator({
+					objectFolderLabel: objectFolder.label['en_US'],
+				})
+			).toBeVisible();
+		}
+
+		// Clean up
+
+		for (const objectFolder of objectFolders) {
+			await apiHelpers.objectAdmin.deleteObjectFolder(objectFolder.id);
+		}
+	});
 });
 
-test('default folder does not contains delete and edit options', async ({
-	viewObjectDefinitionsPage,
-}) => {
-	await viewObjectDefinitionsPage.goto();
+test.describe('Manage object definitions through ViewObjectDefinitions', () => {
+	test('created object folders are on the left side bar', async ({
+		apiHelpers,
+		viewObjectDefinitionsPage,
+	}) => {
+		await viewObjectDefinitionsPage.goto();
+
+		const objectFolderExternalReferenceCode =
+			'objectFolder' + getRandomInt();
+
+		const objectFolder = await viewObjectDefinitionsPage.createObjectFolder(
+			objectFolderExternalReferenceCode
+		);
+
+		await expect(
+			viewObjectDefinitionsPage.page
+				.locator('li')
+				.filter({hasText: objectFolderExternalReferenceCode})
+		).toBeVisible();
+
+		// Clean up
+
+		await apiHelpers.objectAdmin.deleteObjectFolder(objectFolder.id);
+	});
+
+	test('default folder does not contains delete and edit options', async ({
+		viewObjectDefinitionsPage,
+	}) => {
+		await viewObjectDefinitionsPage.goto();
+
+		await viewObjectDefinitionsPage.clickDefaultObjectFolder();
+
+		await viewObjectDefinitionsPage.openObjectFolderActions();
+
+		await expect(
+			viewObjectDefinitionsPage.objectFolderDeleteFolderOption
+		).toBeHidden();
+
+		await expect(
+			viewObjectDefinitionsPage.objectFolderEditLabelAndERCOption
+		).toBeHidden();
+	});
+
+	test('navigate between object folders by ViewObjectDefinitions', async ({
```

Comment:
same explanation as above, should be `navigate between object folders on ViewObjectDefinitions`. also, here you used `ViewObjectDefinitions` (the name of the file) and on the other test you used `Model Builder` (the name of the page/feature). 

i think how we name the tests should have a pattern, and since the `describes` already has the `ViewObjectDefinitions` name, no need to repeat here in my opinion. 

you could use `navigate between object folders on object definitions initial page` or something else following this line; the name of the test should reflect more the action the user is performing, and they don't know the file name of the page they are using. do you agree with me? we can discuss about this if you want

Reply:
Hey rebeca, here I am using `ViewObjectDefinitions` because it is the name of the page (playwright page) that represents this initial page of the list of object definitions.

I believe we can repeat the name of the page to avoid having two tests with the same name.

In this case we have two similar tests that aim to navigate through the folders but on two different pages, I thought about using the page pattern to differentiate the name of the tests:

1. navigate between object folders on ViewObjectDefinitionsPage
2. navigate between object folders on ModelBuilderPage

We can also choose to use the sentence pattern as you mentioned, so we would have "model builder page" and "view object definitions page", for example.

But as this will be a pattern that will be repeated a lot, I would like to hear the opinion of other team members.

Reply:
yes, i agree that is better to ask everyone what they think about this :blush:

Reply:
Hey @cgmonte @danvb112, we would like to hear your opinion :microphone:

Reply:
Hey, I agree with @rebsilva that we shouldn't use the page name as Model Builder and View Object Definitions in all tests if we already said that in `test.describe`, it's ambiguous, I think the `test.describe` should say the page we are testing as ModelBuilder and ViewObjectDefinitions, and the test should only contain what we are testing within this page.

for example if I want to edit the details of a Folder object in ViewObjectDefinitinos, it would be:

```tsx
test.describe('Manage object definitions through ViewObjectDefinitions', () => {
    test('can edit object folder label and ERC', async ({
        ...
     })
}
```

What do you think?

Reply:
What if we are testing the same action. As in the example mentioned:

navigate between object folders on ViewObjectDefinitionsPage
navigate between object folders on ModelBuilderPage

What strategies should we use to avoid repeating the name of the test? Since we can run the test by name in the terminal.

Reply:
I think we can keep the same name, as it's the same test but in different places, and the `test.describe` will determine which place it is.

Reply:
I'm going to reproduce this scenario with more than one test with the same name. And check if we have any problems :shipit:

Reply:
I will disagree with everyone on this. `ViewObjectDefinitions` is a source file. I don't think we should refer to the application's source files in functional tests, as the rendered UI screens for a given test will probably have many source files involved. This page has a name and it is "Objects". I know is not the best name and we should probably rename to Object Definitions, and well. Here is my proposed test description:

`'Manage object folders through Objects page'`

and then

`'navigate between object folders inside Objects page`

Reply:
I don't see a problem with having `Objects page` or `Model Builder` at the end of each test title. It will facilitate running specific tests depending on what you want to test.

Reply:
Update, I can find this [issue](https://github.com/microsoft/playwright/issues/13969) in playwright repo. But I can't able to reproduce the same problem.

But I'm not comfortable with having two tests with the same name.

And about:

> "View Object Definitions is a source file."

I imagined that we had agreed together that we would call this page ViewObjectDefinitions in the last meeting we had. This is already merged into the master as the name of the Playwrite Page.

Reply:
:gem:  as we discussed in person, the format will be like this:

- the whole text in lower case
- when the test is very similar to other, add where it is happening (the page)
- when writing the page, use the format with lower case and spaces: `view object definition`
- use the connectors `inside` or `on `when referencing a page 

one example following this format:
`'navigate between object folders inside view object definitions page'`